### ISSUE TITLE
Getting "remote: Repository not found." error during multi-checkout

### ISSUE THREAD

**Issue by joelshea** [microsoft/azure-pipelines-yaml#447]
We are encountering a very strange error.  When we run a pipeline with multi-checkout, we get the following error:

```
Starting: Checkout `devops` repo
==============================================================================
Task         : Get sources
Description  : Get sources from a repository. Supports Git, TfsVC, and SVN repositories.
Version      : 1.0.0
Author       : Microsoft
Help         : [More Information](https://go.microsoft.com/fwlink/?LinkId=798199)
==============================================================================
Syncing repository: <our-org>/devops (github)
git version
git version 2.25.0
git lfs version
git-lfs/2.9.2 (GitHub; linux amd64; go 1.13.1)
git init "/home/vsts/work/1/s"
Initialized empty Git repository in /home/vsts/work/1/s/.git/
git remote add origin https://github.com/<our-org>/devops
git config gc.auto 0
git config --get-all http.https://github.com/<our-org>/devops.extraheader
git config --get-all http.proxy
git -c http.extraheader="AUTHORIZATION: basic ***" fetch --force --tags --prune --progress --no-recurse-submodules origin
remote: Repository not found.
fatal: repository 'https://github.com/<our-org>/devops/' not found
##[warning]Git fetch failed with exit code 128, back off 7.01 seconds before retry.
git -c http.extraheader="AUTHORIZATION: basic ***" fetch --force --tags --prune --progress --no-recurse-submodules origin
remote: Repository not found.
fatal: repository 'https://github.com/<our-org>/devops/' not found
##[warning]Git fetch failed with exit code 128, back off 5.159 seconds before retry.
git -c http.extraheader="AUTHORIZATION: basic ***" fetch --force --tags --prune --progress --no-recurse-submodules origin
remote: Repository not found.
fatal: repository 'https://github.com/<our-org>/devops/' not found
##[error]Git fetch failed with exit code: 128
Finishing: Checkout `devops` repo
```

The weird part is that this repository is where our templates are also stored and even in the same template:

```yaml
name: $(SourceBranchName)_$(Date:yyyyMMdd)$(Rev:.r)
trigger:
  - master
  - develop
  - feature/*
resources:
  repositories:
    - repository: devops
      type: github
      name: <our-org>/devops
      endpoint: <our-org>

extends:
  template: pipeline-templates/main.yml@devops
  parameters:
    imagename: filter-service
    buildTemplate:
      - template: build-gradle.yml
        parameters:
          jdkVersion: 1.8

# main.yml
variables:
  - group: Shared Variables

parameters:
- name: imagename
  type: string
- name: buildTemplate
  type: step

stages:
- stage: build
  jobs:
  - job: 
    displayName: Build and Push Result
    pool:
      name: Hosted Ubuntu 1604
    steps:
    - ${{ parameters.buildTemplate }}
    - template: build-docker.yml
      parameters:
        imagename: ${{ parameters.imagename }}
    - template: build-migration.yml
      parameters:
        imagename: ${{ parameters.imagename }}
- stage: feature_branch
  condition: and(succeeded(), startsWith(variables['Build.SourceBranch'], 'refs/heads/feature/'))
  dependsOn: build
  jobs:
  - template: deploy-create-temp-stackfile.yml
    parameters:
      servicename: ${{ parameters.imagename }}
  - template: deploy.yml
    parameters:
      environment: FeatureBranch
      stackfile: ${{ parameters.imagename }}-${{ variables['Build.SourceBranchName'] }}
      swarm: feature 

# deploy-create-temp-stackfile.yml
parameters:
  servicename: ''

jobs:
  - job: Stack_file_creation
    displayName: Create Stack File for deploying feature branch
    pool:
        name: Hosted Ubuntu 1604
    steps:
    - checkout: devops
      displayName: Checkout Devops
      persistCredentials: true # So that we can push to the repo as well.
    - checkout: self
      displayName: Checkout Self
      path: s/source # Give 'self' repository specific path so we can reference it in a consistent way.
     .... and a bunch of bash scripts ...
```

But what makes this even weirder is that the we use these templates for about 5 other templates and they all work without error.   It's just this one.  I have double checked the service connection and it's been set to have no restrictions.  What are we doing wrong?

**Comment by kubacech** [comment]
I had to create second github service connection and use that. Looks like one that was created in pipeline creation works only for that repository. 

Dont know why, but maybe it will help others.

**Comment by joelshea** [comment]
For us, the connection works for many pipelines, but not some others.  Also, it always works to get the templates, it only fails when calling "checkout".  It's a real mystery.

**Comment by maartenplugwise** [comment]
First I tried upgrading the permissions for the Azure Pipeline specifically so it can access the right repositories. Didn't work.

Then I changed the permissions of the Github Azure Pipelines app to allow access to all repositories. Didn't work.

For us it started working when I completely removed the azure pipelines app in GitHub and also removed any projects in Azure pipelines. Then reinstall the azure app in GitHub with access to all permissions and setup all projects again.
